src/sidebar/util/oauth-client.js

```diff
@@ -203,21 +197,22 @@ export default class OAuthClient {
    * Make an `application/x-www-form-urlencoded` POST request.
    *
    * @param {string} url
-   * @param {Object} data - Parameter dictionary
+   * @param {Record<string, string>} data - Parameter dictionary
    */
   _formPost(url, data) {
-    // The `fetch` API has native support for sending form data by setting
-    // the `body` option to a `FormData` instance. We are not using that here
-    // because our test environment has very limited `FormData` support and it
-    // is simpler just to format the data manually.
```

Comment:
I know the implementation has changed here, but without any commenting the subsequent use of `params.toString()` as the `body` for the POST requests feels mysterious. Can you help me understand a little better the structure of the resulting POST?

Reply:
`URLSearchParams.toString()` conveniently uses the `application/x-www-form-urlencoded` serialization as POST-ing a `<form>`, which is encoding that the OAuth spec mandates for this endpoint.